src/commands/mod.rs

```diff
@@ -275,6 +325,46 @@ where
     map.end()
 }
 
+fn deserialize_option_string_not_empty<'de, D>(deserializer: D) -> Result<Option<String>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let opt: Option<String> = Option::deserialize(deserializer)?;
+
+    match opt {
+        None => Ok(None),
+        Some(ref s) if s.is_empty() => Err(serde::de::Error::custom("String cannot be empty")),
+        _ => Ok(opt),
+    }
+}
+
+fn serialize_setting_type<S>(setting_type: &SettingType, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: serde::Serializer,
+{
+    serializer.serialize_str(&setting_type.to_string())
+}
+
+fn deserialize_setting_type<'de, D>(deserializer: D) -> Result<SettingType, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: String = Deserialize::deserialize(deserializer)?;
+
+    let valid_setting_types = SettingType::iter()
+        .map(|t| t.to_string())
+        .collect::<Vec<_>>()
+        .join("\n") + "\n";
+
+    match SettingType::from_str(&s.to_lowercase()) {
+        Ok(setting_type) => Ok(setting_type),
+        Err(_) => Err(serde::de::Error::custom(format!(
+            "Field type should be one of the following:\n{}",
+            valid_setting_types
```

Comment:
Here it seems it will print String\nSecret instead of string\nsecret

Reply:
It fortunately won't because strum serialization is defined with lowercase 
```
pub enum SettingType {
    #[default]
    #[strum(serialize = "string")]
    String,
    #[strum(serialize = "secret")]
    Secret,
}
```

Reply:
![image](https://github.com/Screenly/cli/assets/135842671/a2d80908-55ce-4629-bd62-1bbeb5eca0f2)